### PR TITLE
update GH Action versions

### PR DESCRIPTION
**Pull request by seanctech**
# Contributor Comments

## Pull Request Checklist

Thank you for submitting a contribution to cookiecutter-python!

In order to streamline the review of your contribution we ask that you review and comply with the below requirements:

- [x] If you are adding a dependency, please explain how it was chosen.
- [x] If manual testing is needed in order to validate the changes, provide a testing plan and the expected results.
- [x] Validate that documentation is accurate and aligned to any project updates or additions.

Don't forget our more detailed contribution guidelines
[here](https://github.com/SeisoLLC/operations/blob/main/documents/software-guidelines.md#contributing-to-a-repository).

### CODE REVIEW

File: {{cookiecutter.project_name|replace(" ", "")}}/.github/workflows/commit.yml [hunk 1]
```diff
@@ -59,39 +59,41 @@ jobs:
...
       - name: Set up QEMU for cross-platform emulation
-        uses: docker/setup-qemu-action@v2
+        uses: docker/setup-qemu-action@v3
       - name: Build the image(s)
         run: task -v build
         env:
           PLATFORM: ${{ "{{ matrix.platform }}" }}
       - name: Run the tests
-        run: task -v test
+        run: task -v test -- debug
```
Comment (JonZeolla) [inline {{cookiecutter.project_name|replace(" ", "")}}/.github/workflows/commit.yml:74]: Do we want to keep this on or was it just temporary

File: {{cookiecutter.project_name|replace(" ", "")}}/.github/workflows/commit.yml [hunk 2]
```diff
@@ -158,37 +160,37 @@ jobs:
...
-      - uses: actions/cache@v3
+      - uses: actions/cache@v4
         with:
           path: ~/.local/share/virtualenvs
           key: ${{ "{{ runner.os }}" }}-python-${{ "{{ env.python_version }}" }}-pipenv-${{ "{{ hashFiles('Pipfile.lock') }}" }}
       - name: Install the dependencies
         run: python -m pip install --upgrade pipenv
       - name: Install Task
-        uses: arduino/setup-task@v1
+        uses: arduino/setup-task@v3
```
Comment (JonZeolla) [inline {{cookiecutter.project_name|replace(" ", "")}}/.github/workflows/commit.yml:177]: I don't think v3 exists

File: {{cookiecutter.project_name|replace(" ", "")}}/.github/workflows/release.yml [hunk 3]
```diff
@@ -28,49 +28,51 @@ jobs:
...
       - name: Generate vuln scan results
         run: task -v vulnscan
         env:
           PLATFORM: ${{ "{{ matrix.platform }}" }}
       - name: Upload the vuln scan results
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
         with:
-          name: Vulns
+          name: Vulns-${{ "{{ eng.SANITITZED_PLATFORM }}" }}
```
Comment (JonZeolla) [inline {{cookiecutter.project_name|replace(" ", "")}}/.github/workflows/release.yml:71]: Typo - eng should be env and sanitized is spelt wrong

**Review by JonZeolla — CHANGES_REQUESTED** [review]

**Review by ChiefHolland — APPROVED** [review]
LGTM

**Review by JonZeolla — APPROVED** [review]

**Pull request merged**